File: cv-backend/src/main/java/a/gleb/cv/model/request/WorkExperienceModel.java
package a.gleb.cv.model.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.format.annotation.DateTimeFormat;

@Getter
@Setter
@ToString
public class WorkExperienceModel {

    @JsonProperty("start")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String startDate;
    @JsonProperty("end")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private String endDate;
    @JsonProperty("name-of-company")
    private String companyName;
    @JsonProperty("name-of-specialization")
    private String spec;
}

File: cv-frontend/lib/api/models/work_experience_model.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';

@jsonSerializable
class WorkExperienceModel {
  @JsonProperty(name: "start")
  final String startDate;
  @JsonProperty(name: "end")
  final String endDate;
  @JsonProperty(name: "name-of-company")
  final String companyName;
  @JsonProperty(name: "name-of-specialization")
  final String spec;

  WorkExperienceModel(
      this.startDate, this.endDate, this.companyName, this.spec);

  @override
  String toString() {
    return '{'
        '"start": "$startDate",'
        '"end": "$endDate",'
        '"name-of-company": "$companyName",'
        '"name-of-specialization": "$spec"}';
  }
}
